doc/search-api.md

```diff
@@ -46,176 +59,190 @@ The parameters supported are:
-     The examples are sorted by decreasing popularity.  An example facet value
-     in a response with this option set as "examples:1" might look like:
-
-        "value" => {
-          "slug" => "an-example-facet-slug",
-          "example_info" => {
-            "total" => 3,  # The total number of matching examples
-            "examples" => [
-              {"title" => "Title of the first example", "link" => "/foo"},
-            ],
-          }
-        }
 
-   - `example_scope`: `global` or `query`.  If the `examples` option is supplied, the
-     `example_scope` option must be supplied too.
+## Error reporting
 
-     The value of `global` causes the returned examples to be taken from all
-     documents in which the facet field has the given slug.
+The search API supports many query string parameters.  It validates
+parameters strictly - any unknown parameters, or parameters with invalid
+options, will cause an HTTP 422 error.  This makes it likely that typos do not
+result in silently returning the wrong results, and also makes it easier to
+modify the API to add new features without risking breaking old calls.
 
-     The value of `query` causes the returned examples to be taken only from
-     those documents which match the query (and all filters).
+## Returning specific document fields
 
-   - `example_fields`: colon separated list of fields.
+See the [field reference](https://docs.publishing.service.gov.uk/apis/search/fields.html) for a list of all fields returned by the search API.
+
+Only a few fields are returned by default. You can override the fields returned using the `fields` parameter. For example:
+
+```
+https://www.gov.uk/api/search.json?q=passport&fields=mainstream_browse_pages&fields=title
+```
+
+Note that query parameters which are repeated may be specified in standard HTTP
+style (ie, `name=value&name=value`, where the same name may be used multiple
+times), or in Ruby/PHP array style (ie, `name[]=value&name[]=value`).  The `[]`
+is simply ignored.
 
-     If the examples option is supplied, this lists the fields which are
-     returned for each example.  By default, only a small number of fields are
-     returned for each.
+## Building facetted search with filters and aggregations
 
- - fields: fields to be returned in the result documents.  By default, and for
-   backwards compatibility, a fairly long set of fields is currently returned,
-   but it is good practice to set this to only the fields you actually want
-   information on (doing this will normally increase performance).
+### Filter parameters
+You can filter search results using any of the filter_<field name> or reject_<field name> query parameters.
 
- - ab_tests: a/b test with selected variant type. This allows test to be configured
-   from upstream apps.
+- filter is used to only returns documents that match a value
+- reject excludes documents that don't match a value
```

Comment:
I think this sentence is wrong. 

reject excludes documents that match a value